File: src/main/java/io/felipeandrade/gsw2/material/ColorProviderMaterialItem.java
package io.felipeandrade.gsw2.material;

import net.fabricmc.fabric.api.client.rendering.v1.ColorProviderRegistry;
import net.minecraft.item.ItemStack;

public class ColorProviderMaterialItem extends GSWMaterialItem implements GSWItemColorProvider {


    public ColorProviderMaterialItem(String unlocalizedName, GSWMaterial material, Settings settings) {
        super(unlocalizedName, material, settings);
    }

    @Override
    public int getColor(ItemStack stack, int tintIndex) {
        return material.getPrimaryColor();
    }

    @Override
    public void onClientInit() {
        ColorProviderRegistry.ITEM.register(this, this);
    }
}

File: gsw-mod/src/main/kotlin/io/felipeandrade/gsw/material/ColorProviderMaterialItem.kt
package io.felipeandrade.gsw.material

import net.fabricmc.fabric.api.client.rendering.v1.ColorProviderRegistry
import net.minecraft.item.ItemStack

class ColorProviderMaterialItem(unlocalizedName: String, material: GSWMaterial, settings: Settings) :
    GSWMaterialItem(unlocalizedName, material, settings), GSWItemColorProvider {
    override fun getColor(stack: ItemStack, tintIndex: Int): Int {
        return material.primaryColor
    }

    override fun onClientInit() {
        ColorProviderRegistry.ITEM.register(this, this)
    }
}